amd_tressfx_vulkan/src/TressFXMeshVulkan.cpp

```diff
@@ -0,0 +1,1147 @@
+        fillInitialData(upload_cmd_buffer, scratchBuffer, uploadBuffer,
+                        m_HairAsset.m_pVertices, m_HairVertexPositionsPrevBuffer,
+                        offsetInUploadBuffer, bufferDescUA.size);
+    }
+
+    // -----------------------------
+    // m_HairVertexTangentsUAB
+    //------------------------------
+    {
+        VkBufferCreateInfo bufferDescUA{VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO};
+        bufferDescUA.size = m_HairAsset.m_NumTotalHairVertices * sizeof(XMFLOAT4);
+        bufferDescUA.usage = VK_BUFFER_USAGE_UNIFORM_TEXEL_BUFFER_BIT |
+                             VK_BUFFER_USAGE_STORAGE_BUFFER_BIT |
+                             VK_BUFFER_USAGE_TRANSFER_DST_BIT;
+        //        AMD_SAFE_RELEASE(m_HairVertexTangentsUAB);
+        AMD_V_RETURN(vkCreateBuffer(pvkDevice, &bufferDescUA, nullptr,
+                                    &m_HairVertexTangentsBuffer));
+        m_HairVertexTangentsMemory = allocBufferMemory(
+            pvkDevice, m_HairVertexTangentsBuffer, texture_buffer_memory_index);
+        fillInitialData(upload_cmd_buffer, scratchBuffer, uploadBuffer,
+                        m_HairAsset.m_pTangents, m_HairVertexTangentsBuffer,
+                        offsetInUploadBuffer, bufferDescUA.size);
+    }
+
+    //-----------------------
+    // m_HairVertexPositionsView
+    //-----------------------
+    {
+        VkBufferViewCreateInfo sbSRVDesc{VK_STRUCTURE_TYPE_BUFFER_VIEW_CREATE_INFO};
+        sbSRVDesc.range = m_HairAsset.m_NumTotalHairVertices * 4 * sizeof(float);
+        sbSRVDesc.format = VK_FORMAT_R32G32B32A32_SFLOAT;
+        sbSRVDesc.buffer = m_HairVertexPositionsBuffer;
+        //        AMD_SAFE_RELEASE(m_HairVertexPositionsSRV);
+        AMD_V_RETURN(vkCreateBufferView(pvkDevice, &sbSRVDesc, nullptr,
+                                        &m_HairVertexPositionsView));
+    }
+
+    //-----------------------
+    // m_HairVertexTangentsView
+    //-----------------------
+    {
+        VkBufferViewCreateInfo sbSRVDesc{VK_STRUCTURE_TYPE_BUFFER_VIEW_CREATE_INFO};
+        sbSRVDesc.range = m_HairAsset.m_NumTotalHairVertices * 4 * sizeof(float);
+        sbSRVDesc.format = VK_FORMAT_R32G32B32A32_SFLOAT;
+        sbSRVDesc.buffer = m_HairVertexTangentsBuffer;
+        //        AMD_SAFE_RELEASE(m_HairVertexTangentsSRV);
+        AMD_V_RETURN(vkCreateBufferView(pvkDevice, &sbSRVDesc, nullptr,
+                                        &m_HairVertexTangentsView));
+    }
+
+    //-----------------------
+    // m_HairVertexPositionsView
+    //-----------------------
+    {
+        VkBufferViewCreateInfo sbUAVDesc{VK_STRUCTURE_TYPE_BUFFER_VIEW_CREATE_INFO};
+        sbUAVDesc.range = m_HairAsset.m_NumTotalHairVertices * 4 * sizeof(float);
+        sbUAVDesc.format = VK_FORMAT_R32G32B32A32_SFLOAT;
+        sbUAVDesc.buffer = m_HairVertexPositionsPrevBuffer;
+        //        AMD_SAFE_RELEASE(m_HairVertexPositionsUAV);
```

Comment:
Please remove the `AMD_SAFE_RELEASE` calls when not needed - just noise at this point.